tsdb/head_test.go

```diff
@@ -4744,3 +4744,61 @@ func TestGaugeFloatHistogramWALAndChunkHeader(t *testing.T) {
 
 	checkHeaders()
 }
+
+func TestSnapshotAheadOfWALError(t *testing.T) {
```

Comment:
The current implementation loads the chunk snapshot even if it is old, which makes me believe that the unit test is not doing its job. I have suggested a modification below.